### PR TITLE
feat: add lib.rs

### PR DESCRIPTION
**Pull request by Qu4k**
Even if the api is not currently stable it could be helpful to provide third party access to this crate.

### CODE REVIEW

**Comment by bartlomieju** [comment]
I do intend to release a library crate, but for now we don't even have publish step to crates.io and this project is still in very early stages. @Qu4k thank you for PR but I'm gonna close it without a merge.

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#[macro_use]
+extern crate lazy_static;
+
+mod colors;
+mod diagnostic;
+mod linter;
+mod rules;
+mod scopes;
+mod swc_util;
```
Comment (bartlomieju) [inline src/lib.rs:9]: `src/main.rs` should import from this file

File: src/swc_util.rs [hunk 2]
```diff
@@ -198,3 +198,9 @@ impl AstParser {
   //   }
   // }
 }
+
+impl Default for AstParser {
+  fn default() -> Self {
+    Self::new()
+  }
```
Comment (bartlomieju) [inline src/swc_util.rs:205]: Unrelated change, please revert

File: Cargo.toml [hunk 3]
```diff
@@ -6,6 +6,9 @@ description = "lint for deno"
 authors = ["the Deno authors"]
 license = "MIT"
 
+[lib]
+name = "dlint"
```
Comment (bartlomieju) [inline Cargo.toml:10]: Use `deno_lint`

**Comment by Qu4k** [comment]
@bartlomieju now CI is now passing because the change in `src/swc_util.rs` was reverted.

**Comment by bartlomieju** [comment]
@Qu4k I'm changing CI setup at the moment; I'll get back to you once that's done

**Comment by bartlomieju** [comment]
@Qu4k CI should be done, can you try rebasing?

File: src/swc_util.rs (hunk 2 above)
Comment (bartlomieju) [reply src/swc_util.rs]: Alright, seems it's required

File: src/main.rs [hunk 4]
```diff
@@ -1,21 +1,9 @@
 // Copyright 2020 the Deno authors. All rights reserved. MIT license.
-#![deny(warnings)]
```
Comment (bartlomieju) [inline src/main.rs:2]: This line should stay

File: src/swc_util.rs (hunk 2 above)
Comment (Qu4k) [reply src/swc_util.rs]: If you want I can open another pull for the sake of keeping changes related

Comment (bartlomieju) [reply src/swc_util.rs]: It's fine to include it it this PR

File: src/main.rs (hunk 4 above)
Comment (Qu4k) [reply src/main.rs:2]: this makes clippy angry for the color attributes that are not used

Comment (bartlomieju) [reply src/main.rs:2]: I see, okay, let's remove it for now
